share/zfsnap/core.sh

```diff
@@ -203,6 +203,34 @@ GreaterDate() {
     return 0
 }
 
+#Find common snapshot from two lists of snapshot names
+FindCommonSnapshot(){
+	local local_snaps=$1
+	local remote_snaps=$2
+	
+	#test to see if snapshot names were provided
+	if [[ -n "${local_snaps[*]}" && -n "${remote_snaps[*]}" ]]; then
+		#find common snapshot on remote and local servers
+		#this sometimes chops off after the second latest common snapshot, does not break things
+		local common_snaps=$(echo -e "${remote_snaps[*]}" "\n" "${local_snaps[*]}" | sort | uniq -d)
```

Comment:
In general `echo` is bad for your health and `-e` is non-posix anyway. `printf` would be better here.